Managers/PresetMan.cpp

```diff
@@ -181,37 +156,52 @@ bool PresetMan::LoadDataModule(string moduleName, bool official, ProgressCallbac
-		if (!LoadDataModule(officialModule, true, &LoadingGUI::LoadingSplashProgressReport)) { return false; }
+		if (!LoadDataModule(officialModule, true, &LoadingGUI::LoadingSplashProgressReport)) {
+			return false;
+		}
 	}
 
 	// If a single module is specified, skip loading all other unofficial modules and load specified module only.
-	if (m_SingleModuleToLoad != "Base.rte" && m_SingleModuleToLoad != "") {
-		if (!LoadDataModule(m_SingleModuleToLoad, false, &LoadingGUI::LoadingSplashProgressReport)) { return false; } else { return true; }
+	if (!m_SingleModuleToLoad.empty() && std::find(officialModules.begin(), officialModules.end(), m_SingleModuleToLoad) == officialModules.end()) {
+		if (!LoadDataModule(m_SingleModuleToLoad, false, &LoadingGUI::LoadingSplashProgressReport)) {
+			g_ConsoleMan.PrintString("ERROR: Failed to load DataModule \"" + m_SingleModuleToLoad + "\"! Only official modules were loaded!");
+			return false;
+		}
 	} else {
-		al_ffblk moduleInfo;
-		int moduleID = 0;
-
-		for (int result = al_findfirst("*.rte", &moduleInfo, FA_DIREC | FA_RDONLY); result == 0; result = al_findnext(&moduleInfo)) {
-			if (!g_SettingsMan.IsModDisabled(moduleInfo.name)) {
-				moduleID = GetModuleID(moduleInfo.name);
-				// Make sure we don't load properties of already loaded official modules
-				if (strlen(moduleInfo.name) > 0 && (moduleID < 0 || moduleID >= GetOfficialModuleCount()) && string(moduleInfo.name) != "Metagames.rte" && string(moduleInfo.name) != "Scenes.rte") {
+		for (const std::filesystem::directory_entry &directoryEntry : std::filesystem::directory_iterator(System::GetWorkingDirectory())) {
+			std::string directoryEntryPath = directoryEntry.path().generic_string();
+			if (std::filesystem::is_directory(directoryEntryPath) && std::regex_match(directoryEntryPath, std::regex(".*\.rte"))) {
+				std::string moduleName = directoryEntryPath.substr(directoryEntryPath.find_last_of('/') + 1, std::string::npos);
+				if (!g_SettingsMan.IsModDisabled(moduleName) && (std::find(officialModules.begin(), officialModules.end(), moduleName) == officialModules.end() && moduleName != "Metagames.rte" && moduleName != "Scenes.rte")) {
+					int moduleID = GetModuleID(moduleName);
 					// NOTE: LoadDataModule can return false (especially since it may try to load already loaded modules, which is okay) and shouldn't cause stop, so we can ignore its return value here.
-                    LoadDataModule(string(moduleInfo.name), false, &LoadingGUI::LoadingSplashProgressReport);
+					if (moduleID < 0 || moduleID >= GetOfficialModuleCount()) { LoadDataModule(moduleName, false, &LoadingGUI::LoadingSplashProgressReport); }
 				}
 			}
 		}
-		// Close the file search to avoid memory leaks
-		al_findclose(&moduleInfo);
+		// Load scenes and MetaGames AFTER all other techs etc are loaded; might be referring to stuff in user mods.
+		if (!LoadDataModule("Scenes.rte", false, &LoadingGUI::LoadingSplashProgressReport)) {
```

Comment:
I'm pretty sure these should still be loaded when you tell the game to only load a single module, maybe they wanna play metagame (lol) or do scene stuff with their mod.

Is the problem that these may be referencing content from a module that's not being loaded cause of the flag? Maybe the solution to that is to make it always try to load them, but if m_SingleModuleToLoad is specified, just ignore if they fail to load?

Reply:
Single module loading is more of a debugging tool, if someone wants to play with only one mod they should use the mod manager.

Reply:
Okay, that makes sense. I wonder at the point of it at all though? Does it seem like something you'd actually ever use?

Reply:
I guess?
It's just easier and faster to force loading one module if you're making a mod instead of manually disabling your 9000 mods in the manager and then enabling them back once you're done, or moving them back and forth.

Reply:
Wha? That takes us back to the start, that they probably wanna load metagame saves and custom scenes if they're making a mod, no? E.g. if they're iterating on custom scenes, it's easier to tweak it in that rte, and whenever we make the metagame decent, there'll probably be mod config stuff that needs iterating on too.

Anyway, the point is, these should be loaded, assuming it's not super difficult.

Reply:
Nah I don't care enough to add all that complexity to it.
Best for debugging and mod making if you know what you're doing.

Reply:
Yeah, if it does mean a bunch of extra work then sure, leave it be.